app/search_builders/hyrax/collection_search_builder.rb

```diff
@@ -6,9 +6,55 @@ module Hyrax
   # @see https://github.com/samvera/hyrax/blob/master/app/search_builders/hyrax/README.md SearchBuilders README
   # @note the default_processor_chain defined by Blacklight::Solr::SearchBuilderBehavior provides many possible points of override
   #
+  # Allows :deposit as a valid type
   class CollectionSearchBuilder < ::SearchBuilder
     include FilterByType
 
+    attr_reader :access
+
+    # @overload initialize(scope)
+    #   @param [Object] scope scope the scope where the filter methods reside in.
+    # @overload initialize(processor_chain, scope)
+    #   @param [List<Symbol>,TrueClass] processor_chain options a list of filter methods to run or true, to use the default methods
+    #   @param [Object] scope the scope where the filter methods reside in.
+    # @overload initialize(processor_chain, scope)
+    #   @param [List<Symbol>,TrueClass] processor_chain options a list of filter methods to run or true, to use the default methods
+    #   @param [Object] scope the scope where the filter methods reside in.
+    #   @param [Symbol] access one of :edit, :read, or :deposit
+    def initialize(*options) # rubocop:disable Metrics/MethodLength, Metrics/CyclomaticComplexity
```

Comment:
Instead of overriding the initializer could you override or make use of the `.with_access` method from https://github.com/samvera/hyrax/blob/master/app/search_builders/hyrax/search_filters.rb#L10 ?